ray-curator/ray_curator/stages/classifiers/fineweb_edu.py

```diff
@@ -0,0 +1,357 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from dataclasses import dataclass
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+
+import numpy as np
+import pandas as pd
+import torch
+from transformers import AutoModelForSequenceClassification
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.stages.base import CompositeStage, ProcessingStage
+from ray_curator.stages.classifiers.base import HFModel, HFTokenizerStage
+from ray_curator.stages.modules.score_filter import Filter
+from ray_curator.tasks import DocumentBatch
+
+from .constants import ATTENTION_MASK_COLUMN, DEBERTA_TOKENIZER_PADDING_SIDE, INPUT_ID_COLUMN
+
+FINEWEB_EDU_MODEL_IDENTIFIER = "HuggingFaceFW/fineweb-edu-classifier"
+FINEWEB_MIXTRAL_EDU_MODEL_IDENTIFIER = "nvidia/nemocurator-fineweb-mixtral-edu-classifier"
+FINEWEB_NEMOTRON_EDU_MODEL_IDENTIFIER = "nvidia/nemocurator-fineweb-nemotron-4-edu-classifier"
+MAX_SEQ_LENGTH = 512
+
+
+class HFFineWebModelStage(HFModel):
+    """
+    Stage for Hugging Face model inference.
+
+    Args:
+        model_identifier: The identifier of the Hugging Face model.
+        pred_column: The name of the prediction column.
+        float_score_column: The name of the float score column.
+        int_score_column: The name of the integer score column.
+        micro_batch_size: The size of the micro-batch. Defaults to 256.
```

Comment:
Why did we call this `micro_batch_size` and just not `batch_size` , what is a `micro-batch` ?

Reply:
`batch_size` is already a property for each `ProcessingStage` (representing the number of `DocumentBatch` objects to pass into a stage at a time), so it was a naming issue.

Although maybe `model_batch_size` or `inference_batch_size` may be clearer. I can update.